sycl/plugins/unified_runtime/CMakeLists.txt

```diff
@@ -1,31 +1,63 @@
 # PI Unified Runtime plugin library
 #
-if (NOT DEFINED UNIFIED_RUNTIME_INCLUDE_DIR)
+if (NOT DEFINED UNIFIED_RUNTIME_LIBRARY OR NOT DEFINED UNIFIED_RUNTIME_INCLUDE_DIR)
   include(FetchContent)
 
-  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
-  set(UNIFIED_RUNTIME_TAG fd711c920acc4434cb52ff18b078c082d9d7f44d)
+  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src//unified-runtime.git")
+  set(UNIFIED_RUNTIME_TAG 61cb864b3a73918124bff11f9ad58dafb8c7769a)
 
   message(STATUS "Will fetch Unified Runtime from ${UNIFIED_RUNTIME_REPO}")
   FetchContent_Declare(unified-runtime
     GIT_REPOSITORY    ${UNIFIED_RUNTIME_REPO}
     GIT_TAG           ${UNIFIED_RUNTIME_TAG}
   )
 
-  FetchContent_MakeAvailable(unified-runtime)
+  # Disable some compilation options to avoid errors while building the UR
+  if (UNIX)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-but-set-variable")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-pedantic")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-stringop-truncation")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-parameter")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-suggest-override")
+  endif()
+
   FetchContent_GetProperties(unified-runtime)
+  FetchContent_MakeAvailable(unified-runtime)
 
+  set(UNIFIED_RUNTIME_LIBRARY loader)
   set(UNIFIED_RUNTIME_SOURCE_DIR
     ${unified-runtime_SOURCE_DIR} CACHE PATH "Path to Unified Runtime Headers")
+  set(UNIFIED_RUNTIME_LOADER_SOURCE_DIR
+    ${unified-runtime_SOURCE_DIR}/source/loader CACHE PATH "Path to Unified Runtime Loader Headers")
   set(UNIFIED_RUNTIME_INCLUDE_DIR "${UNIFIED_RUNTIME_SOURCE_DIR}/include")
 endif()
 
+add_library (UnifiedRuntimeLoader INTERFACE)
+get_filename_component(UNIFIED_RUNTIME_LIBRARY_SRC "${UNIFIED_RUNTIME_LOADER_SOURCE_DIR}" DIRECTORY)
+get_filename_component(UNIFIED_RUNTIME_LIB_NAME "${UNIFIED_RUNTIME_LIBRARY}" NAME)
+
+message(STATUS "UNIFIED_RUNTIME_LIB_NAME ${UNIFIED_RUNTIME_LIB_NAME}")
+
 add_library (UnifiedRuntime-Headers INTERFACE)
+add_library (UnifiedRuntimeLoader-Headers INTERFACE)
```

Comment:
do we really need two different headers interfaces here? why?

Reply:
please see above.